src/common/util/select-unit.ts

```diff
@@ -49,29 +53,56 @@ export function selectUnit(
     };
   }
 
-  const days = secs / SECS_PER_DAY;
+  const fromDate = new Date(from);
+  const toDate = new Date(to);
+
+  // Set time component to zero, which allows us to compare only the days
+  fromDate.setHours(0, 0, 0, 0);
+  toDate.setHours(0, 0, 0, 0);
+
+  const days = (+fromDate - +toDate) / MS_PER_SECOND / SECS_PER_DAY;
```

Comment:
Please use `date-fns` `differenceInDays`

Reply:
Done. Please note that the first two lines are still needed since `differenceInDays` counts full days.